File: Strings/RotateString/RotateString.ts
function rotate(s: string, goal: string): boolean {
    if (s.length !== goal.length) {
        return false;
    }
    let doubled = s + s;
    return doubled.includes(goal);
}

// Example usage
console.log(rotate("abcde", "cdeab")); // true
console.log(rotate("abcde", "abced")); // false

File: Strings/RotateString/RotateString.py
def rotate(s: str, goal: str) -> bool:
    if len(s) != len(goal):
        return False
    doubled = s + s
    return goal in doubled


# Example usage
print(rotate("abcde", "cdeab"))  # True
print(rotate("abcde", "abced"))  # False
